roles/privacyidea/templates/nginx_privacyidea.conf.j2

```diff
@@ -0,0 +1,54 @@
+# Make sure that port 80 is open in the IONOS firewall for HTTP traffic.
+# This configuration is necessary to complete the initial SSL certificate setup using Certbot.
+
+#server {
+#    listen 80;
+#    server_name {{ privacyidea_server_name }};
+ 
+#    location /.well-known/acme-challenge/ {
+#        root /var/www/letsencrypt;
+#    }
+ 
+#    location /static {
+#        alias /opt/privacyidea/virtualenv/lib/python3.10/site-packages/privacyidea/static;
+#    }
+    #Redirect all other HTTP requests to HTTPS if  the SSL certificate is obtained
+    #This Block can be removed if no SSL is obtained
+    #location / {
+    #   return 301 https://$host$request_uri;
+    #}
+ 
+#    location / {
+#        include uwsgi_params;
+#        uwsgi_pass unix:/run/uwsgi/app/privacyidea/privacyidea.socket;
+#    }
+#}
+
+# After creating the SSL certificate, you can remove the block above that listens on port 80
```

Comment:
@simoncolincap @aimee-889 ,
When setting up an Nginx web server for the first time, the SSL certificate does not exist initially. Therefore, I can't forward requests from port 80 to 443 at the beginning, **because the path ssl_certificate /etc/letsencrypt/live/ will not exist the first time.**
I didn't fully understand your approach. For me,  the following process is clear:

1. Initial configuration (no SSL certificates):

- Configure Nginx to listen only on port 80
- Open port 80 in the firewall
- Set up configuration for HTTP traffic

2. Generate certificates manually:

- Port 80 must remain open for the ACME challenge
- Use Certbot or a similar tool to create SSL certificates

3. Open port 443:

- Allow HTTPS traffic on port 443 in the firewall
- Activate HTTPS server block:

- Uncomment the prepared HTTPS server block in the Nginx configuration
- Verify the paths to the generated SSL certificates
- Restart Nginx to apply the changes

Reply:
How are the certificates generated at the moment? If possible this should also be automated. Maybe this can be done by using a different Nginx config if the certificates don't exist yet.

Reply:
this should be automated - but can be done later.